tests/enums/test_color.py

```diff
@@ -0,0 +1,295 @@
+from unittest import TestCase
+
+import vapoursynth as vs
+
+from vstools import (
+    ColorRange,
+    Matrix,
+    Primaries,
+    Transfer,
+    UnsupportedMatrixError,
+    UnsupportedPrimariesError,
+    UnsupportedTransferError,
+)
+
+
+class TestMatrix(TestCase):
+    def test_is_unknown(self) -> None:
+        self.assertTrue(Matrix.is_unknown(Matrix.UNKNOWN))
+        self.assertTrue(Matrix.is_unknown(2))
+        self.assertFalse(Matrix.is_unknown(Matrix.RGB))
+        self.assertFalse(Matrix.is_unknown(0))
+
+    def test_from_res_rgb(self) -> None:
+        clip = vs.core.std.BlankClip(format=vs.RGB24)
+        result = Matrix.from_res(clip)
+        self.assertEqual(result, Matrix.RGB)
+
+    def test_from_res_uhd(self) -> None:
+        clip = vs.core.std.BlankClip(format=vs.YUV420P8, width=3840, height=2160)
+        result = Matrix.from_res(clip)
+        self.assertEqual(result, Matrix.BT2020NC)
+
+    def test_from_res_hd(self) -> None:
+        clip = vs.core.std.BlankClip(format=vs.YUV420P8, width=1920, height=1080)
+        result = Matrix.from_res(clip)
+        self.assertEqual(result, Matrix.BT709)
+
+    def test_from_res_sd(self) -> None:
+        clip = vs.core.std.BlankClip(format=vs.YUV420P8, width=640, height=480)
+        result = Matrix.from_res(clip)
+        self.assertEqual(result, Matrix.SMPTE170M)
+
+    def test_from_res_pal(self) -> None:
+        clip = vs.core.std.BlankClip(format=vs.YUV420P8, width=1024, height=576)
+        result = Matrix.from_res(clip)
+        self.assertEqual(result, Matrix.BT470BG)
+
+    def test_from_video_rgb(self) -> None:
+        clip = vs.core.std.BlankClip(format=vs.RGB24)
+        result = Matrix.from_video(clip)
+        self.assertEqual(result, Matrix.RGB)
+
+    def test_from_video_uhd(self) -> None:
+        clip = vs.core.std.BlankClip(format=vs.YUV420P8, width=3840, height=2160)
```

Comment:
the from_video tests should check for the property

Reply:
Added more tests around this